src/gc/config.d

```diff
@@ -190,27 +190,10 @@ bool parse(T:const(char)[])(const(char)[] optname, ref const(char)[] str, ref T
 in { assert(str.length); }
 body
 {
-    bool findNext(string next)
-    {
-        if(next.length>str.length)
-            return false;
-
-        return (next.length == str.length)? (next == str):(isspace(str[next.length+1])&&(next == str[0..next.length]));
-    }
-
-    if(findNext("manual"))
-    {
-        res = "manual";
-    }
-    else if(findNext("conservative"))
-    {
-        res = "conservative";
-    }
-    else
-    {
-        return parseError("conservative or manual", optname, str);
-    }
-
+    // [_a-zA-Z][_a-zA-Z0-9]*]
+    if (str[0] != '_' && !isalpha(str[0]))
+        return parseError("an identifier", optname, str);
+    res = str[0 .. $ - str[1 .. $].find!(c => c != '_' && !isalnum(c)).length];
```

Comment:
We are pretty permissive about characters in option names, we should not be more picky with the name of the GC. I think everything but the same terminators should be accepted (i.e. ' ', ':' and '='). This avoids having to camel-case names composed of multiple words, e.g. `mark-and-sweep`.